synapse/event_auth.py

```diff
@@ -1054,9 +1054,14 @@ def _verify_third_party_invite(
     """
     if "third_party_invite" not in event.content:
         return False
-    if "signed" not in event.content["third_party_invite"]:
+    third_party_invite = event.content["third_party_invite"]
+    if not isinstance(third_party_invite, collections.abc.Mapping):
+        return False
+    if "signed" not in third_party_invite:
+        return False
+    signed = third_party_invite["signed"]
+    if not isinstance(signed, collections.abc.Mapping):
         return False
-    signed = event.content["third_party_invite"]["signed"]
     for key in {"mxid", "token"}:
```

Comment:
Should we be checking for `signatures` here? The [spec says it's required](https://spec.matrix.org/v1.6/client-server-api/#mroommember):


Name | Type | Description
-- | -- | --
mxid | string | Required: The invited matrix user ID. Must be equal to the user_id property of the event.
signatures | Signatures | Required: A single signature from the verifying server, in the format specified by the Signing Events section of the server-server API.
token | string | Required: The token property of the containing third_party_invite object.

We use it below without checking if it exists which seems like it could also cause errors like you're trying to stop here

Reply:
If we use it without checking then sounds like it is required! 👍